File: api/Preferred.Api/Models/ReservationAdmin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Preferred.Api.Models
{
    public class ReservationLookupOptionDto
    {
        public int Id { get; set; }
        public string Label { get; set; } = string.Empty;
        public string? Description { get; set; }
    }

    public class ReservationClubAdminSearchParams
    {
        public string? Keyword { get; set; }
    }

    public class ReservationClubAdminDto
    {
        public int Id { get; set; }
        public string ClubCode { get; set; } = string.Empty;
        public string ClubName { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string? District { get; set; }
        public string? Address { get; set; }
        public string? BusinessHours { get; set; }
        public bool IsActive { get; set; }
        public int SeqNo { get; set; }
        public DateTime CrtTime { get; set; }
        public DateTime UpdTime { get; set; }
    }

    public class ReservationClubAdminEditDto
    {
        [Required]
        [StringLength(50)]
        public string ClubCode { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string ClubName { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string City { get; set; } = string.Empty;

        [StringLength(50)]
        public string? District { get; set; }

        [StringLength(255)]
        public string? Address { get; set; }

        [StringLength(100)]
        public string? BusinessHours { get; set; }

        public bool IsActive { get; set; } = true;
        public int SeqNo { get; set; }
    }

    public class ReservationTrainerAdminSearchParams
    {
        public string? Keyword { get; set; }
        public int? ClubId { get; set; }
        public bool? IsActive { get; set; }
    }

    public class ReservationTrainerAdminDto
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string DisplayName { get; set; } = string.Empty;
        public int ClubId { get; set; }
        public string ClubName { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string Gender { get; set; } = string.Empty;
        public int YearsOfExperience { get; set; }
        public decimal Rating { get; set; }
        public int ReviewCount { get; set; }
        public int ServedClients { get; set; }
        public int Satisfaction { get; set; }
        public decimal BasePrice { get; set; }
        public string? TrainingArea { get; set; }
        public string? Highlight { get; set; }
        public string? Introduction { get; set; }
        public string? Story { get; set; }
        public string? HeroImageUrl { get; set; }
        public string? HeroTone { get; set; }
        public string? AccentTone { get; set; }
        public bool IsRecommended { get; set; }
        public bool IsActive { get; set; }
        public int SeqNo { get; set; }
        public List<string> Goals { get; set; } = new List<string>();
        public List<string> Specialties { get; set; } = new List<string>();
        public List<string> Badges { get; set; } = new List<string>();
        public List<string> Certifications { get; set; } = new List<string>();
        public DateTime CrtTime { get; set; }
        public DateTime UpdTime { get; set; }
    }

    public class ReservationTrainerAdminEditDto
    {
        [Required]
        public int UserId { get; set; }

        [Required]
        public int ClubId { get; set; }

        [Required]
        [StringLength(100)]
        public string DisplayName { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [StringLength(10)]
        public string Gender { get; set; } = string.Empty;

        public int YearsOfExperience { get; set; }
        public decimal Rating { get; set; }
        public int ReviewCount { get; set; }
        public int ServedClients { get; set; }
        public int Satisfaction { get; set; }
        public decimal BasePrice { get; set; }

        [StringLength(100)]
        public string? TrainingArea { get; set; }

        [StringLength(100)]
        public string? Highlight { get; set; }

        [StringLength(1000)]
        public string? Introduction { get; set; }

        [StringLength(1000)]
        public string? Story { get; set; }

        [StringLength(500)]
        public string? HeroImageUrl { get; set; }

        [StringLength(20)]
        public string? HeroTone { get; set; }

        [StringLength(20)]
        public string? AccentTone { get; set; }

        public bool IsRecommended { get; set; }
        public bool IsActive { get; set; } = true;
        public int SeqNo { get; set; }
        public List<string> Goals { get; set; } = new List<string>();
        public List<string> Specialties { get; set; } = new List<string>();
        public List<string> Badges { get; set; } = new List<string>();
        public List<string> Certifications { get; set; } = new List<string>();
    }

    public class ReservationSessionAdminSearchParams
    {
        public int? TrainerProfileId { get; set; }
        public string? Keyword { get; set; }
    }

    public class ReservationSessionAdminDto
    {
        public int Id { get; set; }
        public int TrainerProfileId { get; set; }
        public string TrainerName { get; set; } = string.Empty;
        public string SessionCode { get; set; } = string.Empty;
        public string SessionName { get; set; } = string.Empty;
        public string? Description { get; set; }
        public int DurationMinutes { get; set; }
        public decimal Price { get; set; }
        public bool IsActive { get; set; }
        public int SeqNo { get; set; }
        public DateTime CrtTime { get; set; }
        public DateTime UpdTime { get; set; }
    }

    public class ReservationSessionAdminEditDto
    {
        [Required]
        public int TrainerProfileId { get; set; }

        [Required]
        [StringLength(50)]
        public string SessionCode { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string SessionName { get; set; } = string.Empty;

        [StringLength(255)]
        public string? Description { get; set; }

        public int DurationMinutes { get; set; } = 60;
        public decimal Price { get; set; }
        public bool IsActive { get; set; } = true;
        public int SeqNo { get; set; }
    }

    public class ReservationScheduleAdminSearchParams
    {
        public int? TrainerProfileId { get; set; }
        public int? ClubId { get; set; }
        public string? ScheduleDate { get; set; }
        public bool? IsAvailable { get; set; }
    }

    public class ReservationScheduleAdminDto
    {
        public int Id { get; set; }
        public int TrainerProfileId { get; set; }
        public string TrainerName { get; set; } = string.Empty;
        public int ClubId { get; set; }
        public string ClubName { get; set; } = string.Empty;
        public string ScheduleDate { get; set; } = string.Empty;
        public string StartTime { get; set; } = string.Empty;
        public string EndTime { get; set; } = string.Empty;
        public bool IsAvailable { get; set; }
        public int SeqNo { get; set; }
        public DateTime CrtTime { get; set; }
        public DateTime UpdTime { get; set; }
    }

    public class ReservationScheduleAdminEditDto
    {
        [Required]
        public int TrainerProfileId { get; set; }

        [Required]
        public int ClubId { get; set; }

        [Required]
        public string ScheduleDate { get; set; } = string.Empty;

        [Required]
        public string StartTime { get; set; } = string.Empty;

        [Required]
        public string EndTime { get; set; } = string.Empty;

        public bool IsAvailable { get; set; } = true;
        public int SeqNo { get; set; }
    }

    public class ReservationPackageAdminSearchParams
    {
        public int? MemberId { get; set; }
        public int? ClubId { get; set; }
        public string? StatusCode { get; set; }
    }

    public class ReservationPackageAdminDto
    {
        public int Id { get; set; }
        public int MemberId { get; set; }
        public string MemberName { get; set; } = string.Empty;
        public int ClubId { get; set; }
        public string ClubName { get; set; } = string.Empty;
        public string PackageName { get; set; } = string.Empty;
        public string? MembershipName { get; set; }
        public int TotalSessions { get; set; }
        public int RemainingSessions { get; set; }
        public string EffectiveDate { get; set; } = string.Empty;
        public string ExpireDate { get; set; } = string.Empty;
        public string StatusCode { get; set; } = string.Empty;
        public int SeqNo { get; set; }
        public DateTime CrtTime { get; set; }
        public DateTime UpdTime { get; set; }
    }

    public class ReservationPackageAdminEditDto
    {
        [Required]
        public int MemberId { get; set; }

        [Required]
        public int ClubId { get; set; }

        [Required]
        [StringLength(100)]
        public string PackageName { get; set; } = string.Empty;

        [StringLength(100)]
        public string? MembershipName { get; set; }

        public int TotalSessions { get; set; }
        public int RemainingSessions { get; set; }

        [Required]
        public string EffectiveDate { get; set; } = string.Empty;

        [Required]
        public string ExpireDate { get; set; } = string.Empty;

        [Required]
        [StringLength(20)]
        public string StatusCode { get; set; } = "Active";

        public int SeqNo { get; set; }
    }

    public class ReservationOrderAdminSearchParams
    {
        public int? MemberId { get; set; }
        public int? TrainerProfileId { get; set; }
        public string? StatusCode { get; set; }
    }

    public class ReservationOrderAdminDto
    {
        public int Id { get; set; }
        public string ReservationNo { get; set; } = string.Empty;
        public int MemberId { get; set; }
        public string MemberName { get; set; } = string.Empty;
        public int TrainerProfileId { get; set; }
        public string TrainerName { get; set; } = string.Empty;
        public int ClubId { get; set; }
        public string ClubName { get; set; } = string.Empty;
        public int SessionTypeId { get; set; }
        public string SessionName { get; set; } = string.Empty;
        public string ReservationDate { get; set; } = string.Empty;
        public string StartTime { get; set; } = string.Empty;
        public string EndTime { get; set; } = string.Empty;
        public decimal PriceAmount { get; set; }
        public string StatusCode { get; set; } = string.Empty;
        public string? Remark { get; set; }
        public DateTime CrtTime { get; set; }
        public DateTime UpdTime { get; set; }
    }

    public class ReservationOrderStatusUpdateDto
    {
        [Required]
        [StringLength(20)]
        public string StatusCode { get; set; } = string.Empty;
    }
}

File: ui/preferred-admin/src/api/reservationAdmin.ts
import request from '@/utils/request'

export interface PagedResponse<T> {
  data: T[]
  total: number
  page: number
  pageSize: number
  totalPages: number
}

export interface ApiResponse<T = unknown> {
  success: boolean
  message: string
  data?: T
}

export interface LookupOption {
  id: number
  label: string
  description?: string
}

export interface ReservationClub {
  id: number
  clubCode: string
  clubName: string
  city: string
  district?: string
  address?: string
  businessHours?: string
  isActive: boolean
  seqNo: number
  crtTime: string
  updTime: string
}

export interface ReservationClubEditParams {
  clubCode: string
  clubName: string
  city: string
  district?: string
  address?: string
  businessHours?: string
  isActive: boolean
  seqNo: number
}

export interface ReservationTrainer {
  id: number
  userId: number
  userName: string
  displayName: string
  clubId: number
  clubName: string
  title: string
  gender: string
  yearsOfExperience: number
  rating: number
  reviewCount: number
  servedClients: number
  satisfaction: number
  basePrice: number
  trainingArea?: string
  highlight?: string
  introduction?: string
  story?: string
  heroImageUrl?: string
  heroTone?: string
  accentTone?: string
  isRecommended: boolean
  isActive: boolean
  seqNo: number
  goals: string[]
  specialties: string[]
  badges: string[]
  certifications: string[]
  crtTime: string
  updTime: string
}

export interface ReservationTrainerEditParams {
  userId: number
  clubId: number
  displayName: string
  title: string
  gender: string
  yearsOfExperience: number
  rating: number
  reviewCount: number
  servedClients: number
  satisfaction: number
  basePrice: number
  trainingArea?: string
  highlight?: string
  introduction?: string
  story?: string
  heroImageUrl?: string
  heroTone?: string
  accentTone?: string
  isRecommended: boolean
  isActive: boolean
  seqNo: number
  goals: string[]
  specialties: string[]
  badges: string[]
  certifications: string[]
}

export interface ReservationSessionType {
  id: number
  trainerProfileId: number
  trainerName: string
  sessionCode: string
  sessionName: string
  description?: string
  durationMinutes: number
  price: number
  isActive: boolean
  seqNo: number
  crtTime: string
  updTime: string
}

export interface ReservationSessionTypeEditParams {
  trainerProfileId: number
  sessionCode: string
  sessionName: string
  description?: string
  durationMinutes: number
  price: number
  isActive: boolean
  seqNo: number
}

export interface ReservationScheduleSlot {
  id: number
  trainerProfileId: number
  trainerName: string
  clubId: number
  clubName: string
  scheduleDate: string
  startTime: string
  endTime: string
  isAvailable: boolean
  seqNo: number
  crtTime: string
  updTime: string
}

export interface ReservationScheduleSlotEditParams {
  trainerProfileId: number
  clubId: number
  scheduleDate: string
  startTime: string
  endTime: string
  isAvailable: boolean
  seqNo: number
}

export interface ReservationMemberPackage {
  id: number
  memberId: number
  memberName: string
  clubId: number
  clubName: string
  packageName: string
  membershipName?: string
  totalSessions: number
  remainingSessions: number
  effectiveDate: string
  expireDate: string
  statusCode: string
  seqNo: number
  crtTime: string
  updTime: string
}

export interface ReservationMemberPackageEditParams {
  memberId: number
  clubId: number
  packageName: string
  membershipName?: string
  totalSessions: number
  remainingSessions: number
  effectiveDate: string
  expireDate: string
  statusCode: string
  seqNo: number
}

export interface ReservationOrder {
  id: number
  reservationNo: string
  memberId: number
  memberName: string
  trainerProfileId: number
  trainerName: string
  clubId: number
  clubName: string
  sessionTypeId: number
  sessionName: string
  reservationDate: string
  startTime: string
  endTime: string
  priceAmount: number
  statusCode: string
  remark?: string
  crtTime: string
  updTime: string
}

export interface ReservationListParams {
  page?: number
  pageSize?: number
  keyword?: string
  clubId?: number
  trainerProfileId?: number
  memberId?: number
  statusCode?: string
  isActive?: boolean
  isAvailable?: boolean
  scheduleDate?: string
}

export const reservationAdminApi = {
  getClubOptions(): Promise<LookupOption[]> {
    return request.get('/api/reservationadmin/lookups/clubs')
  },
  getTrainerOptions(): Promise<LookupOption[]> {
    return request.get('/api/reservationadmin/lookups/trainers')
  },
  getUserOptions(keyword?: string): Promise<LookupOption[]> {
    return request.get('/api/reservationadmin/lookups/users', { params: { keyword } })
  },

  getClubList(params: ReservationListParams): Promise<PagedResponse<ReservationClub>> {
    return request.get('/api/reservationadmin/clubs/list', { params })
  },
  getClubDetail(id: number): Promise<ReservationClub> {
    return request.get(`/api/reservationadmin/clubs/${id}`)
  },
  createClub(data: ReservationClubEditParams): Promise<ApiResponse> {
    return request.post('/api/reservationadmin/clubs', data)
  },
  updateClub(id: number, data: ReservationClubEditParams): Promise<ApiResponse> {
    return request.put(`/api/reservationadmin/clubs/${id}`, data)
  },
  deleteClub(id: number): Promise<ApiResponse> {
    return request.delete(`/api/reservationadmin/clubs/${id}`)
  },

  getTrainerList(params: ReservationListParams): Promise<PagedResponse<ReservationTrainer>> {
    return request.get('/api/reservationadmin/trainers/list', { params })
  },
  getTrainerDetail(id: number): Promise<ReservationTrainer> {
    return request.get(`/api/reservationadmin/trainers/${id}`)
  },
  createTrainer(data: ReservationTrainerEditParams): Promise<ApiResponse> {
    return request.post('/api/reservationadmin/trainers', data)
  },
  updateTrainer(id: number, data: ReservationTrainerEditParams): Promise<ApiResponse> {
    return request.put(`/api/reservationadmin/trainers/${id}`, data)
  },
  deleteTrainer(id: number): Promise<ApiResponse> {
    return request.delete(`/api/reservationadmin/trainers/${id}`)
  },

  getSessionList(params: ReservationListParams): Promise<PagedResponse<ReservationSessionType>> {
    return request.get('/api/reservationadmin/sessions/list', { params })
  },
  getSessionDetail(id: number): Promise<ReservationSessionType> {
    return request.get(`/api/reservationadmin/sessions/${id}`)
  },
  createSession(data: ReservationSessionTypeEditParams): Promise<ApiResponse> {
    return request.post('/api/reservationadmin/sessions', data)
  },
  updateSession(id: number, data: ReservationSessionTypeEditParams): Promise<ApiResponse> {
    return request.put(`/api/reservationadmin/sessions/${id}`, data)
  },
  deleteSession(id: number): Promise<ApiResponse> {
    return request.delete(`/api/reservationadmin/sessions/${id}`)
  },

  getScheduleList(params: ReservationListParams): Promise<PagedResponse<ReservationScheduleSlot>> {
    return request.get('/api/reservationadmin/schedules/list', { params })
  },
  getScheduleDetail(id: number): Promise<ReservationScheduleSlot> {
    return request.get(`/api/reservationadmin/schedules/${id}`)
  },
  createSchedule(data: ReservationScheduleSlotEditParams): Promise<ApiResponse> {
    return request.post('/api/reservationadmin/schedules', data)
  },
  updateSchedule(id: number, data: ReservationScheduleSlotEditParams): Promise<ApiResponse> {
    return request.put(`/api/reservationadmin/schedules/${id}`, data)
  },
  deleteSchedule(id: number): Promise<ApiResponse> {
    return request.delete(`/api/reservationadmin/schedules/${id}`)
  },

  getPackageList(params: ReservationListParams): Promise<PagedResponse<ReservationMemberPackage>> {
    return request.get('/api/reservationadmin/packages/list', { params })
  },
  getPackageDetail(id: number): Promise<ReservationMemberPackage> {
    return request.get(`/api/reservationadmin/packages/${id}`)
  },
  createPackage(data: ReservationMemberPackageEditParams): Promise<ApiResponse> {
    return request.post('/api/reservationadmin/packages', data)
  },
  updatePackage(id: number, data: ReservationMemberPackageEditParams): Promise<ApiResponse> {
    return request.put(`/api/reservationadmin/packages/${id}`, data)
  },
  deletePackage(id: number): Promise<ApiResponse> {
    return request.delete(`/api/reservationadmin/packages/${id}`)
  },

  getOrderList(params: ReservationListParams): Promise<PagedResponse<ReservationOrder>> {
    return request.get('/api/reservationadmin/orders/list', { params })
  },
  updateOrderStatus(id: number, statusCode: string): Promise<ApiResponse> {
    return request.put(`/api/reservationadmin/orders/${id}/status`, { statusCode })
  }
}

export default reservationAdminApi
